CorinneTest/Factory.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CorinneTest
+{
+    class Factory : IPersonFactory
+    {
+       
+        static Random rnd = new Random();
+
+
+        public static List<Person> CreateInRange(int lowAge,
+                                                 int highAge,
+                                                 int? numOfPeople)
+        {
+            var list = new List<Person>();
+            int i = 0;
+            while (i < numOfPeople)
+            {
+                int type = rnd.Next(1, 9);
+                if (type == 1) { Person newPerson = MakeWoman(lowAge, highAge); }
+                else if (type == 2) { Person newPerson = MakeSomeone(lowAge, highAge); }
+                else if (type == 3) { Person newPerson = MakeMan(lowAge, highAge); }
+                else if (type == 4) { Person newPerson = MakeGirl(lowAge, highAge); }
+                else if (type == 5) { Person newPerson = MakeBoy(lowAge, highAge); }
+                else if (type == 6) { Person newPerson = MakeChild(lowAge, highAge); }
+                else if (type == 7) { Person newPerson = MakeSeniorWoman(lowAge, highAge); }
+                else if (type == 8) { Person newPerson = MakeSeniorMan(lowAge, highAge); }
+                else if (type == 9) { Person newPerson = MakeSeniorPerson(lowAge, highAge); }
+                list.Add(newPerson);
+            }
+                
+            return list;
+        }
+
+        public static DateTime? GenRanAdultBirthDate(int lowAge, int highAge)
+        {
+            if (highAge < 18)
+                return null;
+            DateTime start = new DateTime((2018 - highAge), 5, 29);
+            DateTime minDate = new DateTime((2018 - lowAge), 5, 29);
+            int range = (minDate - start).Days;
+            return start.AddDays(rnd.Next(range));
```

Comment:
cool logic.

as a slight C#-ism, the right-hand side of a declaration will infer to the compiler the left-hand type.  while its clean to specify DateTime or int on the left, you could just say "var" since the right-hand side is explicit that it will be a DateTime or int or whatever

Reply:
changed to var - good to know.